pkg/ebpf/driver_interface.go

```diff
@@ -0,0 +1,222 @@
+	binary.LittleEndian.PutUint64(buf, signature)
+	return buf
+}
+
+// DriverInterface holds all necessary information for interacting with the windows driver
+type DriverInterface struct {
+	driverHandle windows.Handle
+	iocp         windows.Handle
+	path         string
+}
+
+// NewDriverInterface returns a DriverInterface struct for interacting with the driver
+func NewDriverInterface() (*DriverInterface, error) {
+	dc := &DriverInterface{
+		path: driverFile,
+	}
+
+	handle, err := dc.NewDriverHandle()
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "error creating driver handle", err)
+	}
+
+	// Create IO Completion port that we'll use to communicate with the driver
+	iocp, err := windows.CreateIoCompletionPort(handle, windows.Handle(0), 0, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create IO completion port %v", err)
+	}
+
+	dc.driverHandle = handle
+	dc.iocp = iocp
+
+	// Set the packet filters that will determine what we pull from the driver
+	// TODO: Determine failure condition for not setting filter
+	// TODO: I.e., one or more, all fail? I.E., at what point do we not create a tracer
+	err = dc.prepareDriverFilters()
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup packet filters on the driver: %v", err)
+	}
+
+	return dc, nil
+}
+
+// NewDriverHandle creates a new driver handle attached to the windows driver
+func (dc *DriverInterface) NewDriverHandle() (windows.Handle, error) {
+	p, err := windows.UTF16PtrFromString(dc.path)
+	if err != nil {
+		return windows.InvalidHandle, err
+	}
+	log.Debug("Creating Driver handle...")
+	h, err := windows.CreateFile(p,
+		windows.GENERIC_READ|windows.GENERIC_WRITE,
+		windows.FILE_SHARE_READ|windows.FILE_SHARE_WRITE,
+		nil,
+		windows.OPEN_EXISTING,
+		windows.FILE_FLAG_OVERLAPPED,
+		windows.Handle(0))
+	if err != nil {
+		return windows.InvalidHandle, err
+	}
+	log.Info("Connected to driver and handle created")
+	return h, nil
+}
+
+// CloseDriverHandle closes an open handle on the driver
+func (dc *DriverInterface) CloseDriverHandle(handle windows.Handle) error {
+	return windows.CloseHandle(handle)
+}
+
+// Add buffers to Tracer object. Even though the windows API will actually
+// keep the buffers, add it to the tracer so that golang doesn't garbage collect
+// the buffers out from under us
+func (dc *DriverInterface) prepareReadBuffers(bufs []readBuffer) ([]readBuffer, error) {
+	for i := 0; i < totalReadBuffers; i++ {
+		err := windows.ReadFile(dc.driverHandle, bufs[i].data[:], nil, &bufs[i].ol)
+		if err != nil {
+			if err != windows.ERROR_IO_PENDING {
+				fmt.Printf("failed to initiate readfile %v\n", err)
+				windows.CloseHandle(dc.iocp)
+				dc.iocp = windows.Handle(0)
+				return nil, err
+			}
+		}
+	}
+	return bufs, nil
+}
+
+func (dc *DriverInterface) prepareDriverFilters() (err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		log.Errorf("Error getting interfaces: %s\n", err.Error())
+		return
```

Comment:
I don't think using a named return helps readability for this function.  Can this be `return err`